### ISSUE TITLE
Remove user from space

### ISSUE THREAD

**Issue by KarimChabrak** [Coinsence/meta#125]
When trying to remore user from space, I get the follwing error. 

![image](https://user-images.githubusercontent.com/41925899/66770222-55f42400-eeb7-11e9-8d44-728db245ff10.png)

**Comment by gdalyy** [comment]
I'll fix that asap, it seems to be an error coming from ethereum , @haythem96 what does this mean : 

```
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]: { nonce: 1,
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   gasPrice: BigNumber { _hex: '0x3b9aca00' },
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   gasLimit: BigNumber { _hex: '0x4c4b40' },
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   to: '0xb705b833b2A6413e778c45A4499EE1c048875BF5',
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   value: BigNumber { _hex: '0x00' },
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   data: '0xa9059cbb000000000000000000000000ba2cbda934df329c3cafad665af9c222d463392
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   chainId: 4,
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   v: 44,
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   r: '0xcd9f9e57fd0346864eb6f143909f33f148b789c0f8f38aea86676e5aa7f07c65',
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   s: '0x0d9ded12f8f898bffa5ff22743af951f141cf5d9fa68a904bbb591fe9423ebc1',
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   from: '0x44154eB5972F0E7A659C0eb2C2a47D0C852DF195',
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   hash: '0x1fbcb7741fac675c0aeac76477b7b00c12ed9e600da8f980a7a62bb0c9bc5b6a',
oct. 14 11:14:39 scw-coinsence-production eth-node-community[21008]:   wait: [Function] }

```

**Comment by haythem96** [comment]
do we have a function to remove other space user ?!!!!!!

**Comment by haythem96** [comment]
There is only `leaveSpace()` in the contracts which get called through the `POST /leave` in the API, the user need to call that function to leave a space, it is LEAVE not REMOVE by the admin.

Also if you will change that or you will add leave function in humhub please do not call that endpoint for now...

And I think that's not an error it is just a tx hash.

**Comment by gdalyy** [comment]
Didn't get it ? I'm i calling the wrong endpoint or what ? the `leave()` call is already implemented on humhub side and was working , I don't know what's going wrong this time, i'll try to have a look asap

**Comment by haythem96** [comment]
1. There is a `leave()` endpoint: that function should be called by the user who want to leave a space.
2. There is no function in the smart contract nor an endpoint in the API to remove someone from the space (e.g space admin remove user1)
3. Even if you are trying to implement a `leave` functionality (a user leave the space by his own will), please do not call the `leave` endpoint because the function need improvement,  and below I will explain more... but if you want u can call it.

In the smart contract, there is an array of members addresses `members`, when a member call the leave function using his address, I get the position of that address inside the array and I remove it, but the length of the array does not decrease and in that case we will still have an empty slot.

**Comment by KarimChabrak** [comment]
Dear @haythem96  and @gdalyy 

One question: Does it make sense at the moment to map space members to the blockchain?

At the moment space members have no collective decision power on the blockchain (and also not yet in our platform). So, from blockchain perspective, beeing as space member does not mean anything at the moment.  Of could this will change when we implement DAO functionalities enabling member to (vote for special roles or approve special function like minting coins......)   

So, I think, for the moment we do not need to add any space member to the space DAO.

What do you think?

**Comment by haythem96** [comment]
yes it totally make sense to map members addresses in the space contract, we definitely need that.
